### PR TITLE
Demo of `Links` module: List all missing links from all links on given page(s).

### PR DESCRIPTION
**Pull request by SupriyaSupugade**
From “[Help:Link color](https://en.wikipedia.org/wiki/Help:Link_color)” ,what I believe is Red link is link to a page that are not currently exist within Wikipedia. 

[API:Links](https://www.mediawiki.org/wiki/API:Links) -- GET Request to find all the links on the provided page(s). This module can be used as a generator.

In my solution I used API:Links as generator.

@srish  Please review changes, if it is correct I will proceed and implement UI to demonstrate.

### CODE REVIEW

**Comment by srish** [comment]
Thanks for working on this! Give me 1-2 days time to review your work!

File: modules.json [hunk 1]
```diff
@@ -367,4 +367,19 @@
             "ucuser": "Jimbo Wales"
     	}
     }
-]
\ No newline at end of file
+
```
Comment (srish) [inline modules.json]: No blank line here too. You are missing a comma between two data sets.

File: modules.json [hunk 2]
```diff
@@ -367,4 +367,19 @@
...
+    {
+
+        "filename": "get_alllinks_red.py",
+        "docstring": "List all the links and check if the link is a red link or not.",
```
Comment (srish) [inline modules.json]: Description could be something like: "Demo of `Links` module: Extract missing links from all links on a give page...". Feel free to rephrase but we want it to be consistent with other doc strings in `modules.json`.

File: modules.json [hunk 3]
```diff
@@ -367,4 +367,19 @@
...
+        "endpoint": "https://en.wikipedia.org/w/api.php",
+        "params": {
+                "format": "json",
+    		"action": "query",
+    		"generator":"links",
+    		"titles": "Title",
+    		"gpllimit": "max",
```
Comment (srish) [inline modules.json]: For the purpose of demo app, you could fetch only ~20 pages and use a page that may have many missing links on it already like:
https://en.wikipedia.org/wiki/Wikipedia:Requested_articles/Arts_and_entertainment#Design. You can explore more here: https://en.wikipedia.org/wiki/Wikipedia:Requested_articles.

File: python/get_alllinks_red.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    "generator":"links",
+    "titles": "Title",
+    "gpllimit": "max",
+    "prop": "links"
+}
+
+R = S.get(url=URL, params=PARAMS)
+DATA = R.json()
+
+print(DATA)
```
Comment (srish) [inline python/get_alllinks_red.py]: You will need to process this data further to fetch missing links. Also, make sure to update the title and description in this file too.

**Comment by SupriyaSupugade** [comment]
Thank you for your valuable feedback. I made required changes and added new commit. Please review.

**Comment by SupriyaSupugade** [comment]
@srish I have made required changes according to your review and made new commit. Please review.

**Comment by SupriyaSupugade** [comment]
@srish API Documentation is ready. I modify example part of the API:Links page. You can view documentation on https://www.mediawiki.org/wiki/User:Supriya_Supugade/Sandbox/API:Links this link. Please review and let me know if any changes required. Thank you :)

**Comment by srish** [comment]
@SupriyaSupugade Thanks for your work on this! I've merged @brendajerop's pull request so we can close this one and you can continue to work on other tasks. I would like to emphasize that your contribution was of good quality as well, but as multiple people were working on this, I had to accept the contribution on a first come first serve basis :) 

Even though I didn't merge your pull request; you can still add this as one of the contributions you made in your Outreachy application.

**Comment by SupriyaSupugade** [comment]
@srish Thank you so much for your feedback and considering this task for apllication. I will continue with other task. :)

**Pull request closed without merging**